rust/src/delta_datafusion.rs

```diff
@@ -163,9 +213,51 @@ impl TableProvider for delta::DeltaTable {
     }
 }
 
-fn stat_to_val(stat: &ColumnCountStat) -> Option<usize> {
-    match stat {
-        ColumnCountStat::Value(val) => Some(*val as usize),
-        ColumnCountStat::Column(_) => None,
+fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::ScalarValue> {
+    if stat_val.is_number() {
+        if stat_val.is_i64() {
+            Some(ScalarValue::from(stat_val.as_i64()?))
+        } else if stat_val.is_u64() {
+            Some(ScalarValue::from(stat_val.as_u64()?))
+        } else if stat_val.is_f64() {
+            Some(ScalarValue::from(stat_val.as_f64()?))
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+fn left_larger_than_right(
+    left: datafusion::scalar::ScalarValue,
+    right: datafusion::scalar::ScalarValue,
+) -> bool {
+    match left {
+        ScalarValue::Float64(Some(v)) => {
+            let f_right = f64::try_from(right).unwrap();
+            v > f_right
+        }
+        ScalarValue::Float32(Some(v)) => {
+            let f_right = f32::try_from(right).unwrap();
+            v > f_right
+        }
+        ScalarValue::Int8(Some(v)) => {
+            let i_right = i8::try_from(right).unwrap();
+            v > i_right
+        }
+        ScalarValue::Int16(Some(v)) => {
+            let i_right = i16::try_from(right).unwrap();
+            v > i_right
+        }
+        ScalarValue::Int32(Some(v)) => {
+            let i_right = i32::try_from(right).unwrap();
+            v > i_right
+        }
+        ScalarValue::Int64(Some(v)) => {
+            let i_right = i64::try_from(right).unwrap();
+            v > i_right
+        }
+        _ => unimplemented!("Unimplemented for {:?}", left),
```

Comment:
nitpick, could we make this error message more explicit for easier debugging?

```suggestion
        _ => unimplemented!("Scalar value comparison unimplemented for {:?} and {:?}", left, right),
```